### PR TITLE
feat: Phase 3 Llada2DefaultRemaskingPolicy (issue #7)

### LINKED ISSUE #7: [MVP LLaDA2.0]: Implement remasking/llada2_default.py (MVP default policy)
## Context
Ship one default policy (e.g. confidence threshold + remask remainder) per [DESIGN_MVP.md §8](https://github.com/vllm-project/dllm-plugin/blob/main/docs/DESIGN_MVP.md) and plugin-plan **Position remasking / confidence** direction.

## Acceptance criteria
- [ ] `remasking/llada2_default.py` implements `RemaskingPolicy`.
- [ ] Unit-testable without full vLLM stack where possible.
- [ ] Documented tuning knobs (thresholds) at MVP level.

## Out of scope
Multiple policies per architecture; grammar-gated remasking.

## Depends on
- #6 (RemaskingPolicy contract).

### PR DESCRIPTION
**Pull request by TomerG711**
## Summary

- Add `Llada2DefaultRemaskingPolicy` (`vllm_dllm_plugin.remasking.llada2_default`): per-position argmax with **stable softmax** confidence; commit when prob ≥ threshold, else `mask_token_id` in `next_input_block`. Validates `input_block` / logits shape `(DRAFT_SIZE, vocab_size)`, duck-typed logits (no `torch` import in policy), first-index tie-break on equal top logits.
- Config defaults `LLADA2_DEFAULT_MASK_TOKEN_ID` and `LLADA2_DEFAULT_COMMIT_CONFIDENCE_THRESHOLD` (tuned for Phase 2 mock logits per `docs/MOCK_STACK_MODEL.md`).
- Tests (`tests/test_llada2_default.py`), `CONTRACTS.md` / `DESIGN_MVP.md` cross-refs, export from `remasking` package.

**Context:** [vllm-project/dllm-plugin#7](https://github.com/vllm-project/dllm-plugin/issues/7), Phase 3 in [#19](https://github.com/vllm-project/dllm-plugin/issues/19); design [DESIGN_MVP.md §8](https://github.com/vllm-project/dllm-plugin/blob/main/docs/DESIGN_MVP.md); upstream hook discussion [vllm#36155](https://github.com/vllm-project/vllm/issues/36155).

## Milestone Checklist (MVP LLaDA2.0)

- [x] Issue + phase declared (example: `Related: #20`, `Phase 0`)
- [x] HARD dependencies listed and status noted
- [x] SOFT dependencies listed and status noted
- [x] In-scope work stated
- [x] Out-of-scope work stated
- [x] Validation evidence included (tests/manual checks)
- [x] Docs impact stated (updated or N/A with reason)

### Issue and Phase

- Related issue(s): #7, #13 (follow-up only)
- Closing keyword (only if fully resolved): Closes #7
- Phase: 3

### Dependencies

- HARD: #6 (`RemaskingPolicy` / `RemaskStepResult` on `main`) — satisfied
- SOFT: #16 (additional contract tests may land later); #13 deferred to a **separate** Phase 3 PR (model→policy bridge)

### Scope

- In scope: `llada2_default.py`, config defaults, `remasking` export, unit tests, CONTRACTS/DESIGN pointers, tie-break fix commit
- Out of scope: #13 bridge module, `DllmWorker` / #10, scheduler #8, `validation.py` #4

### Validation Evidence

- Commands run: `uv run pytest -v`, `uv run pre-commit run --all-files`
- Result: 24 passed, 4 skipped (vLLM/torch `importorskip`); pre-commit passed

### Docs Impact

- Files updated: `docs/CONTRACTS.md`, `docs/DESIGN_MVP.md` (§8), `vllm_dllm_plugin/remasking/base.py` (module docstring pointer)

**AI-assisted:** Substantive implementation and tests were developed with Cursor / automated tooling; validated with local pytest and pre-commit.

### CODE REVIEW

File: vllm_dllm_plugin/remasking/llada2_default.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                next_tokens.append(token_id)
+            else:
+                next_tokens.append(mask_token_id)
+
+        result = RemaskStepResult(
+            committed_token_ids=tuple(committed),
+            next_input_block=tuple(next_tokens),
+        )
+        validate_remask_step_result(result)
+        return result
```
Comment (AlonKellner-RedHat) [inline vllm_dllm_plugin/remasking/llada2_default.py:243]: This logic is incorrect, it does not match the LLaDA2.0 logic.
Also, the terminology does not match the design docs.
Here are the discrepancies:

- `input_block` should be named `input_draft`
- Decoded and masked `input_draft` tokens should be treated differently, decoded should not be tested for a threshold
- There should be a "draft completed" path, where when the block is fully decoded - all tokens are committed at the same time, and `next_tokens` is set to a fresh [MASK] block. Tokens should be committed in that path only, and not in the "draft in progress" path.
- When there are no tokens above the threshold, the threshold should be set to the max value (`threshold = max(threshold, max(conf_list))`)

Worth comparing to [the HF LLaDA2.0 remasking implementation](https://huggingface.co/inclusionAI/LLaDA2.0-mini/blob/main/modeling_llada2_moe.py#L1366-L1383).

**Review by AlonKellner-RedHat — APPROVED** [review]
LGTM
